internal/smi/smi.go

```diff
@@ -0,0 +1,272 @@
+package smi
```

Comment:
Should this package be defined in `learn-layer5` instead?

Reply:
No lee, but it should be here as well. This should be a part of the adaptor library that we talked about.